### DISCUSSION TITLE
LES with multi component systems

### DISCUSSION THREAD

**Discussion by fra-kd** [AMReX-Combustion/PeleC#838] [Q&A]
Why I get **LES is not supported for multi-component systems** activating a Constant Smagorisky LES model ? Obviously I'm running a case with two species and the relative mechanism but I do not understand why this limitation! I checked both LES.H and LES.cpp scripts together with the relative PelePhysics routines and the properties seem to be correctly computed for the mixture and do not enter in the turbulence model!

**Comment by baperry2** [comment]
There are some limitations with the LES implementation in PeleC and these warnings are there to ensure that users do not apply LES models to systems where the implementation is incomplete without knowing this. Most notably, the LES implementation includes turbulent fluxes for energy and momentum, but not for species. Additionally, with nonlinearity in thermochemical properties, several additional unclosed terms appear in the LES equations. For example, computing filtered temperature from filtered density, energy, and species is not a closed operation with non-constant specific heats, and no additional treatment is implemented for these additional unclosed terms in PeleC. For say, a mixture of N2 and O2 with no major temperature variation, that approximation is fine. For reacting flows with large temperature variation, it is perhaps not. You can see more details in the notes at the top of LES.cpp. 

If you are aware of these limitations and do not believe they are important in your simulations, there should be no issues with applying the LES models in PeleC. In that case, the LES terms should work fine as implemented and you can ignore the warnings or simply comment them out in your code.

**Reply by fra-kd** [reply]
Thanks so much for your answer Bruce, really appreciated! I'm studying a mixture of H2 and N2 so I think my only issue would be the inclusion of the turbulent fluxes in the species equations... From the online documentation I saw that the diffusive terms are included in the specie equation so, basically, I "just" need to implement the turbulent stresses as source terms (as done for momentum and energy)? Am I right?

**Reply by baperry2** [reply]
Yeah if you want turbulent fluxes for species, you'd have to modify the functions encapsulated by `getLESTerm` to also include species fluxes where momentum and energy fluxes are computed. If you do this, please consider submitting a pull request as this would be handy capability for other users as well.

**Reply by fra-kd** [reply]
Sure ! I will do it ! but let me ask you another question… within the variable `flx` can I define the flux term for specie equations by just changing the second parameter ? Like  `flx(iv, UFS + n)` ? Thanks again

**Reply by baperry2** [reply]
Yes, that should be correct.

**Reply by fra-kd** [reply]
Great! Thanks again!

**Comment by pure2082** [comment]
Sorry to bother you, but may I ask if it is feasible to include subgrid stresses in the species equations? Were you later successful in implementing this?

**Reply by fra-kd** [reply]
Hi! You're welcome! Yes, it is feasible. You have to modify the governing equation by inserting the subgrid stresses, supposing a value of the Schmidt number.

**Comment by bronise777** [comment]
Sorry to bother you, If the SGS flux of the component is added, are there any cases that can be used to verify this modification? In addition, if this LES method is adopted, how is the filtered reaction rate calculated

**Comment by fra-kd** [comment]
Hi @bronise777 ! I didn't try this implementation with reactive flows. Sorry for that! Regarding test cases, I cannot help you. I just assumed a fixed value of the Schmidt number
